ukat/vessels/phase_contrast.py

```diff
@@ -0,0 +1,288 @@
+Dinah S. Khatir, Frank G. Zöllner, Susan T. Francis, Nicholas M. Selby,
+Andrea Remuzzi, Anna Caroli
+
+Magnetic Resonance Materials in Physics, Biology and Medicine (2020) 33:3-21
+https://doi.org/10.1007/s10334-019-00772-0
+"""
+
+import os
+import pandas as pd
+import numpy as np
+import nibabel as nib
+import matplotlib.pyplot as plt
+from tabulate import tabulate
+from ukat.utils.tools import convert_to_pi_range
+
+
+class PhaseContrast:
+    """
+    Generates velocity and flow measurements of an MRI Phase Contrast Sequence.
+
+    Attributes
+    ----------
+    velocity_array : np.ndarray
+        The input velocity array masked.
+    shape : tuple
+        The shape of the velocity array.
+    pixel_spacing : tuple
+        The pixel spacing of the acquisition estimated from the affine.
+    mask : np.ndarray
+        A boolean mask of the voxels to fit.
+    num_pixels : np.ndarray
+        List containing the number of True values in the mask per phase.
+    area : np.ndarray
+        List containing the area (cm2) of the mask per phase.
+    min_velocity : np.ndarray
+        List containing the minimum velocity values (cm/s) per phase.
+    mean_velocity : np.ndarray
+        List containing the average velocity values (cm/s) per phase.
+    max_velocity : np.ndarray
+        List containing the maximum velocity values (cm/s) per phase.
+    std_velocity : np.ndarray
+        List containing the std dev of the velocity values (cm/s) per phase.
+    rbf : np.ndarray
+        List containing the Renal Blood Flow values (ml/min) per phase.
+    mean_velocity_global : float
+        Average velocity (cm/s) accross the different phases.
+    mean_rbf : float
+        Average Renal Blood Flow (ml/min) accross the different phases.
+    resistive_index : float
+        A prognostic marker in renal vascular diseases which range is [0, 1].
+    """
+
+    def __init__(self, velocity_array, affine, mask=None):
+        """Initialise a PhaseContrast class instance.
+
+        Parameters
+        ----------
+        velocity_array : np.ndarray
+            A 3D array containing the velocity images of the phase contrast
+            sequence, i.e. the dimensions of the array are [x, y, p], where p
+            corresponds to the phase (or trigger delay).
+        affine : np.ndarray
+            A matrix giving the relationship between voxel coordinates and
+            world coordinates.
+        mask : np.ndarray, optional
+            A boolean mask of the voxels to fit. Should be the shape of the
+            the raw data.
+        """
+        self.shape = velocity_array.shape
+        self.affine = affine
+        self.pixel_spacing = (np.linalg.norm(self.affine[:3, 1]),
+                              np.linalg.norm(self.affine[:3, 0]))
+        # Generate a mask if there isn't one specified
+        if mask is None:
+            self.mask = np.ones(self.shape, dtype=bool)
```

Comment:
I can't think of any time you wouldn't need a vessel mask and therefore maybe it shouldn't be optional. Any thoughts @charlotteebuchanan?